build/builtin_actors.go

```diff
@@ -1,27 +1,263 @@
 package build
 
 import (
-	"bytes"
+	"archive/tar"
+	"context"
+	"embed"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
 
+	"github.com/filecoin-project/lotus/blockstore"
 	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/actors/adt"
 
-	"github.com/BurntSushi/toml"
+	"github.com/DataDog/zstd"
+	"github.com/ipfs/go-cid"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	"github.com/ipld/go-car"
+	"golang.org/x/xerrors"
 )
 
-var BuiltinActorReleases map[actors.Version]Bundle
+//go:embed actors/*.tar.zst
+var embeddedBuiltinActorReleases embed.FS
 
 func init() {
-	BuiltinActorReleases = make(map[actors.Version]Bundle)
+	if BundleOverrides == nil {
```

Comment:
Just for my understanding, this still runs on every invocation of the Lotus CLI (and shed and others), yes? But is fairly quick if you're just using the embedded bundle?

Reply:
> Just for my understanding, this still runs on every invocation of the Lotus CLI

Yes.

> But is fairly quick if you're just using the embedded bundle

Unless you specify bundle overrides, it doesn't even load the bundles. In that case, it skips:

https://github.com/filecoin-project/lotus/blob/f01bd41cc811beadd2c6d974d0219ce06307d67c/build/builtin_actors.go#L61-L73

And runs:

https://github.com/filecoin-project/lotus/blob/f01bd41cc811beadd2c6d974d0219ce06307d67c/build/builtin_actors.go#L76-L90

Which uses the pre-extracted bundle metadata without actually loading the bundle itself.